### ISSUE TITLE
Errors on rails 4.2.3

### ISSUE THREAD

**Issue by darrenleeweber** [sul-dlss/triannon#206]
.gemspec is pinned to rails 4.2.2 (actually just actionpack 4.2.2 is the issue), because errors arise in 4.2.3, i.e.

```
1) Triannon::AnnotationsController#create creates a new annotation from the body of the request
     Failure/Error: post :create, ttl_data, anno_root: @root_container
     NoMethodError:
       undefined method `_run_save_callbacks' for #<Triannon::Annotation:0x0000000b00b5d0>
     # ./app/models/triannon/annotation.rb:43:in `save'
     # ./app/controllers/triannon/annotations_controller.rb:84:in `create'
     # /home/travis/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/metal/implicit_render.rb:4:in `send_action'
```

**Comment by darrenleeweber** [comment]
Note that fixing the rails gem to 4.2.2 in .gemset did not fix the travis builds, so it's not entirely clear that 4.2.2 vs. 4.2.3 is the source of the problems that motivated this issue in the first place.

**Comment by darrenleeweber** [comment]
The Annotation class might need to be updated for rails 4.2.3, see
  - http://api.rubyonrails.org/classes/ActiveModel/Callbacks.html

**Comment by darrenleeweber** [comment]
`bundle update` would not upgrade to rails 4.2.3 because the `spec/internal/Gemfile` had pinned it.  To get `bundle update` to work properly, try `rm -rf spec/internal; bundle update; rake engine_cart:generate`.  This does not impact travis builds, because `spec/internal` is not in the git repo.

**Comment by darrenleeweber** [comment]
@cbeer - there is a possible conflict in the engine_cart:regenerate process when it is run using rake at the `triannon` project path, because, at some point during the regeneration, there is exists two Gemfile files, one in `triannon/Gemfile` and another in `triannon/spec/internal/Gemfile` and they may conflict with one another, e.g.
```
Bundler could not find compatible versions for gem "tilt":
  In snapshot (Gemfile.lock):
    tilt (= 2.0.1)

  In Gemfile:
    sass-rails (~> 5.0) ruby depends on
      tilt (~> 1.1) ruby

    tilt (>= 0) ruby

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```
(BTW, this error above only appears when `triannon/Gemfile` is hacked to specify a specific tilt dependency, so you have to uncomment that hack to see it.)

**Comment by darrenleeweber** [comment]
This might be fixed by https://github.com/sul-dlss/triannon/commit/407f85c89259b71a814ea329f6d5369f71213d0d

**Comment by darrenleeweber** [comment]
Migrating engine cart concerns into https://github.com/cbeer/engine_cart/issues/23

**Comment by darrenleeweber** [comment]
@ndushay also committed a fix in e50baad4341be504a7682990ae90b276af69c61b

**Comment by darrenleeweber** [comment]
The fixes have passed a build in travis, so closing this issue as fixed (mucho fixed).
